staking-pool/src/lib.rs

```diff
@@ -4,50 +4,73 @@ mod internal;
 mod core_impl;
 mod enumeration;
 
-use near_sdk::collections::LookupMap;
+use near_sdk::collections::{UnorderedSet,LookupMap};
 use near_sdk::{near_bindgen, AccountId, env, PanicOnDefault, Balance, EpochHeight, BlockHeight, BorshStorageKey, Promise, PromiseResult, PromiseOrValue, ext_contract};
-use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize,};
 use near_sdk::serde::{Deserialize, Serialize};
 use near_sdk::json_types::{U128, U64};
 
 use crate::modules::account::{Account, AccountJson, UpgradableAccount};
-use crate::modules::tier::{TierMinPointConfigs, Tier};
+use crate::modules::tier::{TierMinPointConfig, Tier};
 pub use crate::enumeration::PoolInfo;
-use crate::Tier::Tier0;
 use crate::util::*;
 
 
 pub const NO_DEPOSIT: Balance = 0;
 pub const DEPOSIT_ONE_YOCTOR: Balance = 1;
 pub const NUM_EPOCHS_TO_UNLOCK: EpochHeight = 1;
 
-#[derive(BorshDeserialize, BorshSerialize, Deserialize, Serialize, Clone, Copy)]
+#[derive(BorshDeserialize, BorshSerialize, Deserialize, Serialize, Clone)]
 #[serde(crate = "near_sdk::serde")]
 pub struct Config {
     // Percent reward per 1 block
-    pub reward_numerator: u32,
+    pub reward_numerator: u64,
     /// What is?
     pub reward_denumerator: u64,
-    pub total_apr: u32,
+    pub total_apr: u64,
 
     /// the config for each user Tier
-    pub tier_point_config: TierMinPointConfigs,
+    pub tier_point_configs: Vec<TierMinPointConfig>,
 }
 
 impl Config {
-    pub fn get_default_tier_min_point_cfg() -> TierMinPointConfigs {
-        let mut cfg = TierMinPointConfigs::new(StorageKey::PointConfigKey);
-        cfg.insert(&Tier::Tier0, &0);
-        cfg.insert(&Tier::Tier1, &100);
-        cfg.insert(&Tier::Tier2, &1_000);
-        cfg.insert(&Tier::Tier3, &5_000);
-        cfg.insert(&Tier::Tier4, &10_000);
+    pub fn get_default_tier_min_point_cfg() -> Vec<TierMinPointConfig> {
+       
+        let mut cfg = Vec::new();
+        cfg.push(TierMinPointConfig{
+            tier: Tier::Tier0,
+            min_point: 0
+        });
+        cfg.push(TierMinPointConfig{
+            tier: Tier::Tier1,
+            min_point: 100
+        });
+        cfg.push(TierMinPointConfig{
+            tier: Tier::Tier2,
+            min_point: 1_000
+        });
+        cfg.push(TierMinPointConfig{
+            tier: Tier::Tier2,
+            min_point: 5_000
+        });
+        cfg.push(TierMinPointConfig{
+            tier: Tier::Tier3,
+            min_point: 10_000
+        });
 
         return cfg;
     }
 
     pub fn set_tier_min_point_cfg(&mut self, tier: Tier, min_point: u64) {
-        self.tier_point_config.insert(&tier, &min_point);
+        let tier_min_point_configs = TierMinPointConfig{
+            tier: tier,
+            min_point: min_point
+        };
+        self.tier_point_configs.push(tier_min_point_configs);
```

Comment:
Using a vector & pushing a new item like that will cause the items to be duplicated when this function has been called multi-time for a specific Tier. Should use the old structure of Unordermap<Tier, TierConfig> or have to handle the duplication here.